internal/wasmldr/src/loader/compiled/tls.rs

```diff
@@ -0,0 +1,340 @@
+
+    async fn get_filetype(&self) -> Result<FileType, Error> {
+        self.any.get_filetype().await
+    }
+
+    async fn get_fdflags(&self) -> Result<FdFlags, Error> {
+        self.any.get_fdflags().await
+    }
+
+    async fn set_fdflags(&mut self, fdflags: FdFlags) -> Result<(), Error> {
+        self.any.set_fdflags(fdflags).await
+    }
+
+    async fn get_filestat(&self) -> Result<Filestat, Error> {
+        self.any.get_filestat().await
+    }
+
+    async fn set_filestat_size(&self, size: u64) -> Result<(), Error> {
+        self.any.set_filestat_size(size).await
+    }
+
+    async fn advise(&self, offset: u64, len: u64, advice: Advice) -> Result<(), Error> {
+        self.any.advise(offset, len, advice).await
+    }
+
+    async fn allocate(&self, offset: u64, len: u64) -> Result<(), Error> {
+        self.any.allocate(offset, len).await
+    }
+
+    async fn set_times(
+        &self,
+        atime: Option<wasi_common::SystemTimeSpec>,
+        mtime: Option<wasi_common::SystemTimeSpec>,
+    ) -> Result<(), Error> {
+        self.any.set_times(atime, mtime).await
+    }
+
+    async fn read_vectored<'a>(&self, bufs: &mut [IoSliceMut<'a>]) -> Result<u64, Error> {
+        let (cap, tls) = &mut *self.lck.write().unwrap();
+
+        if tls.wants_read() {
+            tls.read_tls(cap.deref_mut())?;
+            tls.process_new_packets()?;
+        }
+
+        let n = match tls.reader().read_vectored(bufs) {
+            Err(e) if e.kind() == ErrorKind::UnexpectedEof => 0,
+            Err(e) => return Err(e.into()),
+            Ok(n) => n,
+        };
+
+        Ok(n as u64)
+    }
+
+    async fn read_vectored_at<'a>(
+        &self,
+        _bufs: &mut [IoSliceMut<'a>],
+        _offset: u64,
+    ) -> Result<u64, Error> {
+        Err(Error::badf())
+    }
+
+    async fn write_vectored<'a>(&self, bufs: &[IoSlice<'a>]) -> Result<u64, Error> {
+        let (cap, tls) = &mut *self.lck.write().unwrap();
+
+        let n = tls.writer().write_vectored(bufs)?;
+
+        while tls.wants_write() {
+            tls.write_tls(cap.deref_mut())?;
+        }
+
+        Ok(n as u64)
+    }
+
+    async fn write_vectored_at<'a>(
+        &self,
+        _bufs: &[IoSlice<'a>],
+        _offset: u64,
+    ) -> Result<u64, Error> {
+        Err(Error::badf())
+    }
+
+    async fn seek(&self, pos: std::io::SeekFrom) -> Result<u64, Error> {
+        self.any.seek(pos).await
+    }
+
+    async fn peek(&self, _buf: &mut [u8]) -> Result<u64, Error> {
+        Err(Error::badf())
+    }
+
+    async fn num_ready_bytes(&self) -> Result<u64, Error> {
+        self.any.num_ready_bytes().await
+    }
+
+    fn isatty(&self) -> bool {
+        self.any.isatty()
+    }
+
+    async fn readable(&self) -> Result<(), Error> {
+        self.any.readable().await
+    }
+
+    async fn writable(&self) -> Result<(), Error> {
+        self.any.writable().await
+    }
+}
+
+pub struct Listener {
+    cap: Forgotten<CapListener>,
+    any: AnyListener,
```

Comment:
Neither the field name, nor import name this is renamed to make it clear what is the purpose of this.
At the very least there should be a comment here.
Also, I'm not `AnyListener` rename helps anything here, IMO just using fully-qualified name would be way clearer here